### ISSUE TITLE
Question about compiling with and without IPv6 support

### ISSUE THREAD

**Issue by aldenml** [arvidn/libtorrent#3104]
libtorrent: `master`

What are the real advantages of compiling without IPv6 support these days? I believe all errors are handled if the underlying OS does not support it. Does it make sense drop support for only IPv4?

**Comment by arvidn** [comment]
some systems don't provide the types for IPv6 addresses and defines for `AF_INET6` etc.
I think there may be reasons to disable IPv6 on linux too, under certain circumstances, but I'm not sure I know them off hand.

**Comment by aldenml** [comment]
I underestand that's the underlying implicit assumption (the existence of systems with no IPv6), but I'm under the impression, that a system with not capability to compile with IPv6 should not be supported by this powerful  and networking intensive library. Still, faulty or non functional IPv6 should be handled at runtime.

This should be of great benefit to the source code.

**Comment by arvidn** [comment]
yeah, it's a good point. Disabling using IPv6 on a system that supports it should be done at runtime by setting listen- and outgoing- interfaces.

Systems that don't support IPv6 should still implement the full asio API, just fail with `not_supported`.

**Comment by aldenml** [comment]
Are you open to it (the removal of "only IPv4") for this version (`1.2`) or rather the next one?

**Comment by arvidn** [comment]
I think I am open to that, yes. I should run some tests in the environment where I know libtorrent is currently built with IPv6 to make sure the configuration options have the same effect

**Comment by aldenml** [comment]
Hi @ssiloti, I would like to hear your opinion on this, thanks

**Comment by ssiloti** [comment]
I agree with removing the IPV6 compile time option. I suspect that the intersection of systems which lack any support for IPv6 and systems on which it makes sense to run libtorrent is nil. As @aldenml says, dropping the option would let us remove a lot of ugly boilerplate from the source.

**Comment by arvidn** [comment]
@aldenml I just noticed the arm travis build has IPv6 disabled. iirc you set that up, do you recall why IPv6 was disabled?
